lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -426,6 +428,51 @@ def citations(self, trans: GalaxyWebTransaction, id, **kwds):
             rval.append(citation.to_dict('bibtex'))
         return rval
 
+    @expose_api
+    def conversion(self, trans: GalaxyWebTransaction, tool_id, payload, **kwd):
+        converter = self._get_tool(tool_id, user=trans.user)
+        target_type = payload.get("target_type")
+        source_type = payload.get("source_type")
+        input_src = payload.get("src")
+        input_id = payload.get("id")
+        # List of string of dependencies
+        try:
+            deps = trans.app.datatypes_registry.converter_deps[source_type][target_type]
+        except KeyError:
+            deps = {}
+        # Generate parameter dictionary
+        params = {}
+        # determine input parameter name and add to params
+        input_name = 'input1'
+        for key, value in converter.inputs.items():
+            if deps and value.name in deps:
+                params[value.name] = deps[value.name]
+            elif value.type == 'data':
+                input_name = key
+
+        params[input_name] = {
+            "values": [
+                {
+                    "id": input_id,
+                    "src": input_src,
+                }
+            ],
+            "batch": input_src == "hdca",
+        }
+
+        history_id = payload.get('history_id')
+        if history_id:
+            decoded_id = self.decode_id(history_id)
+        else:
+            encoded_id = int(trans.app.security.decode_id(input_id))
+            decoded_id = self.hda_manager.get_accessible(encoded_id, trans.user).id
```

Comment:
This should be something more like ``self.hda_manager.get_owned(encoded_id, trans.user).history_id``.

You don't want the ID of an HDA, you want the ID of a history. Also we should verify the history is owned I guess.